packages/runtime/container-runtime/src/summary/summarizerNode/summarizerNode.ts

```diff
@@ -193,24 +194,95 @@ export class SummarizerNode implements IRootSummarizerNode {
 	}
 
 	/**
-	 * Complete the WIP summary for the given proposalHandle
+	 * Validates that the in-progress summary is correct, i.e., summarize should have run for all non-skipped
+	 * nodes. This will only be called for the root summarizer node and is called by it recursively on all child nodes.
+	 *
+	 * @returns ValidateSummaryResult which contains a boolean success indicating whether the validation was successful.
+	 * In case of failure, additional information is returned indicating type of failure and where it was.
+	 */
+	public validateSummary(): ValidateSummaryResult {
+		return this.validateSummaryCore(false /* parentSkipRecursion */);
+	}
+
+	/**
+	 * Validates that the in-progress summary is correct for all nodes, i.e., summarize should have run for all
+	 * non-skipped nodes.
+	 * @param parentSkipRecursion - true if the parent of this node skipped recursing the child nodes when summarizing.
+	 * In that case, the children will not have work-in-progress state.
+	 *
+	 * @returns ValidateSummaryResult which contains a boolean success indicating whether the validation was successful.
+	 * In case of failure, additional information is returned indicating type of failure and where it was.
+	 */
+	protected validateSummaryCore(parentSkipRecursion: boolean): ValidateSummaryResult {
+		assert(
+			this.wipSummaryLogger !== undefined,
+			"wipSummaryLogger should have been set in startSummary or ctor",
+		);
+		assert(this.wipReferenceSequenceNumber !== undefined, "Not tracking a summary");
+
+		// If the parentSkipRecursion is true, the parent node did not call summarize on this node. Return success.
+		if (parentSkipRecursion) {
+			return { success: true };
```

Comment:
This codepath used to also check `this._latestSummary !== undefined` and in that case it would invoke the recursive call on its children.  Now it just returns.

Hmmm... I guess since parentSkipRecursion will be true in those recursive calls... it'll just be a big no-op.  Makes sense.  Does that path have test coverage?  I would assume so due to existing `completeSummaryCore` coverage?

Reply:
Nice -_- (sarcastic)

Reply:
I believe Nicholas and I both agreed this is unnecessary but we are scared to remove it because we don't have enough test coverage :D